tools/topology/m4/detect.m4

```diff
@@ -0,0 +1,55 @@
+divert(-1)
+
+dnl Define macro for generic detection widget
+
+dnl Detect name)
+define(`N_DETECT', `DETECT'PIPELINE_ID`.'$1)
+
+dnl W_DETECT(name, format, periods_sink, periods_source, detect_type, mixer_list, enum_list)
+define(`W_DETECT',
+`SectionVendorTuples."'N_DETECT($1)`_tuples_w" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."word" {'
+`		SOF_TKN_COMP_PERIOD_SINK_COUNT'		STR($3)
+`		SOF_TKN_COMP_PERIOD_SOURCE_COUNT'	STR($4)
+`	}'
+`}'
+`SectionVendorTuples."'N_DETECT($1)`_detect_process_tuples_str" {'
+`	tokens "sof_process_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_PROCESS_TYPE'	STR(concat($5, _DETECT))
+`		SOF_TKN_PROCESS_SUBTYPE	"SIGSINK"'
```

Comment:
@keyonjie i discussed with @plbossart about this and we should use the snd_soc_dapm_sink type for the detect component. This way we can avoid the subtype. In the kernel this widget will use the same ipc struct as the process to keep changes minimal. So lets remove this token.

Reply:
@ranj063 per your latest comments in kernel PR, we can't use snd_soc_dapm_sink as it is with always on PM, so let me resolve this conversation.